relay-profiling/src/cocoa.rs

```diff
@@ -151,10 +210,17 @@ pub fn parse_cocoa_profile(payload: &[u8]) -> Result<Vec<u8>, ProfileError> {
         return Err(ProfileError::NotEnoughSamples);
     }
 
-    match serde_json::to_vec(&profile) {
-        Ok(payload) => Ok(payload),
-        Err(_) => Err(ProfileError::CannotSerializePayload),
+    if profile.transactions.is_empty() && !profile.has_transaction_metadata() {
+        return Err(ProfileError::NoTransactionAssociated);
     }
+
+    for transaction in profile.transactions.iter() {
+        if !transaction.clone().valid() {
```

Comment:
Here too, changing the `valid` method to accept a reference to the instance here we don't need to clone each transactions to check their validity.